test/test.js

```diff
@@ -106,16 +152,284 @@ test('esLoader module', function(t){
+  });
+
+  handleCsv(txtFile, badTxtRecord, scratchSpace, function(){
+  },
+  function(record, err){
+    t.ok(err, 'Errors without a spatial reference.');
+  });
+
+  handleCsv('fake', csvRecord, scratchSpace, function(){
+  },
+  function(record, err){
+    t.ok(err, 'Fails with bad filename.');
+  });
+});
+
+
+
+
+test('unzipFile module', function(t){
+ t.plan(3);
+ var zipfile = 'test/data/arkansas.zip';
+ var badfile = 'test/data/nofilehere.notzip';
+ var record = {name: 'arkansas', file: 'arkansas.zip'};
+ var badRecord = {name: 'arkansas'};
+
+ var dir1 = path.join(scratchSpace, crypto.pseudoRandomBytes(10).toString('hex'));
+ var dir2 = path.join(scratchSpace, crypto.pseudoRandomBytes(10).toString('hex'));
+ var dir3 = path.join(scratchSpace, crypto.pseudoRandomBytes(10).toString('hex'));
+
+ unzipFile(zipfile, record, dir1, function(file){
+   t.ok(file, 'Successfully unzips file.');
+ },
+ function(record, err){
+   t.notOk(err);
+ });
+
+ unzipFile(badfile, record, dir2, function(file){
+   t.notOk(file);
+ },
+ function(record, err){
+   t.ok(err, 'Error on bad file.');
+ });
+
+ unzipFile(zipfile, badRecord, dir3, function(file){
+   t.notOk(file);
+ },
+ function(record, err){
+   t.ok(err, 'Error without complete record.');
+ });
+});
+
+
+
+
+test('handleZip module', function(t){
+  t.plan(5);
+
+  var arkRecord = {name: 'arkansas', file: 'arkansas.zip'};
+  var virgRecord = {name: 'virg', file: 'virg.csv', spatialReference: 'NAD83'};
+
+  handleZip(fs.createReadStream('test/data/arkansas.zip'), arkRecord, scratchSpace,
+    function(record, file){
+      t.ok(file, 'handleZip works on zipped shapefile.');
+      t.equal(record, arkRecord, 'Passes through correct record.');
+    }, function(record, err){t.notOk(err)});
+
+  handleZip(fs.createReadStream('test/data/virg.zip'), virgRecord, scratchSpace,
+    function(record, file){
+      t.ok(file, 'handleZip works on zipped csv.');
+      t.equal(record, virgRecord, 'Passes through correct record.');
+    }, function(record, err){t.notOk(err)})
+
+  handleZip(fs.createReadStream('test/data/virginia.csv'), virgRecord, scratchSpace,
+    function(record, file){
+      t.notOk(file);
+    }, function(record, err){t.ok(err, 'Errors on bad file.')})
+});
+
+
+
+
+test('assureRecordCount module', function(t){
+  t.plan(5);
+
+  var countOnly = {count: 123};
+  var deriveCount = {};
+
+  assureRecordCount(deriveCount, 'test/data/arkansas.json', function(err){
+    t.notOk(err, 'Doesn\'t err on good file and args.');
+    t.equal(deriveCount.count, 19, 'Gets the appropriate count.');
+  });
+
+  assureRecordCount(countOnly, 'doesn\'t matter', function(err){
+    t.notOk(err, 'Doesn\'t err on good args and count present.');
+    t.equal(countOnly.count, 123, 'Count remains the same.');
+  });
+
+  assureRecordCount({}, 'test/data/fakebadfile', function(err){
+    t.ok(err, 'Errs on bad file if no count provided.');
+  });
+});
+
+
+
+
+test('ftpWrapper module', function(t){
```

Comment:
These are more like integration tests.  These could start to fail at any time since the servers are out of our control.  Do we want them in the same test suite as the unit tests.  I think you we trying to get a local FTP server running to get around this?